UniversalAlgebra/Varieties/FreeAlgebras.lagda

```diff
@@ -172,10 +172,18 @@ Every `h : X → ∣ 𝑨 ∣` can be decomposed as `h = g ∘ 𝔥`, where `g :
 
    --  eq : (b : B) → (a : A) → b ≡ f a → Image f ∋ b
 
-   γ : h p ≡ h q
-   γ = h (h₀Inv (𝔥₀ p) (eq (𝔥₀ p) p refl)) ≡⟨ cong h ((ζ p q (𝔥₀ q) (pKq ⁻¹) refl )) ⟩
-       h (h₀Inv (𝔥₀ q) (eq (𝔥₀ q) q refl)) ∎
+   γ : h x ≡ h y
+   γ = h (h₀Inv (𝔥₀ x) (eq (𝔥₀ x) x refl)) ≡⟨ cong h ((ζ x y (𝔥₀ y) (xKy ⁻¹) refl )) ⟩
+       h (h₀Inv (𝔥₀ y) (eq (𝔥₀ y) y refl)) ∎
+
+  lemker-ap : ∀ (h : X → ∣ 𝑨 ∣) → kernel (free-lift 𝑨 𝔥₀) ⊆ kernel (free-lift 𝑨 h)
+  lemker-ap h {p , q} pKq = γ
+   where
+   γ : (free-lift 𝑨 h) p ≡ (free-lift 𝑨 h) q
+   γ = (free-lift 𝑨 h) p ≡⟨ {!!} ⟩ (free-lift 𝑨 h) q ∎
 
+  ψlem : ∀ p q → ∣ homℭ ∣ p ≡ ∣ homℭ ∣ q → (free-lift 𝑨 𝔥₀) p ≡ (free-lift 𝑨 𝔥₀) q
+  ψlem p q hyp = {!!}
```

Comment:
Are you sure you want to merge things in that have open interaction points?

Reply:
Yes, this is an experimental branch, but thanks for the comments.